src/test/kotlin/cacophony/utils/FileInputTest.kt

```diff
@@ -0,0 +1,311 @@
+package cacophony.utils
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.io.File
+
+class FileInputTest {
+    private fun createFileWithContent(content: String): String {
+        val file = File.createTempFile("test_file", ".cac")
+        file.deleteOnExit()
+        file.writeText(content)
+        return file.path
+    }
+
+    @Test
+    fun next() {
+        val path = createFileWithContent("abcde")
+        val input = FileInput(path)
+
+        assertEquals('a', input.next())
+        assertEquals('b', input.next())
+        assertEquals('c', input.next())
+        assertEquals('d', input.next())
+        assertEquals('e', input.next())
+        // not checking first character after last - it is '\n'
```

Comment:
I don't understand what you meant here, can you elaborate? Why we don't check the '\n' character?

Reply:
I manually add '\n' to the end of each line, even if it is not present in the file. This is kind of an implementation detail, so there is no assert for that.